datafusion/expr/src/expr_rewriter/order_by.rs

```diff
@@ -116,7 +116,19 @@ fn rewrite_in_terms_of_projection(
 
         // look for the column named the same as this expr
         if let Some(found) = proj_exprs.iter().find(|a| expr_match(&search_col, a)) {
-            return Ok((*found).clone());
+            let found = found.clone();
```

Comment:
The fragile here to me seems to be that the code assumes some naming, and with casting it removes the cast expression. Ideally we wouldn't depend on the naming in the code but on the expressions themselves.

Reply:
Yes I agree this is non ideal -- however I can't think of anything better